pkg/driver/linstor.go

```diff
@@ -251,25 +252,32 @@ func (l *linstor) GetNodes() ([]*storkvolume.NodeInfo, error) {
 	return infos, nil
 }
 
-func (l *linstor) GetPodVolumes(podSpec *v1.PodSpec, namespace string) ([]*storkvolume.Info, error) {
+func (l *linstor) GetPodVolumes(podSpec *v1.PodSpec, namespace string, includePendingWFFC bool) ([]*storkvolume.Info, []*storkvolume.Info, error) {
+	// includePendingWFFC - Includes pending volumes in the second return value if they are using WaitForFirstConsumer binding mode
 	var volumes []*storkvolume.Info
+	var pendingWFFCVolumes []*storkvolume.Info
 	for _, volume := range podSpec.Volumes {
 		volumeName := ""
+		isPendingWFFC := false
 		if volume.PersistentVolumeClaim != nil {
 			pvc, err := core.Instance().GetPersistentVolumeClaim(
 				volume.PersistentVolumeClaim.ClaimName,
 				namespace)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
 			}
 
 			if !l.OwnsPVC(core.Instance(), pvc) {
 				continue
 			}
 
-			if pvc.Status.Phase == v1.ClaimPending {
-				return nil, &storkvolume.ErrPVCPending{
-					Name: volume.PersistentVolumeClaim.ClaimName,
+			if !includePendingWFFC && pvc.Status.Phase == v1.ClaimPending {
+				if includePendingWFFC && isWaitingForFirstConsumer(pvc) {
+					isPendingWFFC = true
```

Comment:
First we check `!includePendingWFFC`, then we guard with `includePendingWFFC`...
I don't know the logic here, but this can never be reached, so `isPendingWFFC` is always false.

Reply:
Yeah, sorry. Just a little copy/paste bug here 😅